apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -821,25 +661,21 @@ const DanceAiModal: React.FunctionComponent<DanceAiModalProps> = ({
         {mode === Mode.EXPLANATION && currentGeneratedEffect && (
           <div id="explanation-area" className={moduleStyles.explanationArea}>
             <div className={moduleStyles.key}>
-              {getRangeArray(0, SLOT_COUNT - 1).map(index => {
-                const item = getItem(inputs[index]);
-                return (
-                  <div key={index} className={moduleStyles.emojiSlot}>
-                    <div
-                      className={classNames(
-                        moduleStyles.dot,
-                        explanationKeyDotColor[index]
-                      )}
-                    />
-                    {item && (
-                      <EmojiIcon
-                        item={item}
-                        className={moduleStyles.emojiSlotIcon}
-                      />
+              {selectedEmojis.map((item, index) => (
+                <div key={item.id} className={moduleStyles.emojiSlot}>
+                  <div
+                    className={classNames(
+                      moduleStyles.dot,
+                      explanationKeyDotColor[index]
```

Comment:
As a general rule, keying off of indexes is frowned upon and it's better to key with a unique value, such as the item's id in this case.

Further, since we now have the `selectedEmojis` list, just iterates on that instead.